server/webapp/WEB-INF/rails.new/spec/new_javascripts/models/pipeline_configs/template/template_spec.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define([
+  'lodash', 'mithril', 'string-plus', 'models/pipeline_configs/template/template', 'models/pipeline_configs/stages'
+], function (_, m, s, Template, Stages) {
+  describe('Template', function () {
+
+    var templateJSON = {
+      name: "template.name",
+      stages: [
+        {
+          name:                    "UnitTest",
+          fetch_materials:         true,
+          clean_working_directory: false,
+          never_cleanup_artifacts: true,
+          environment_variables:   [
+            {
+              name:   "MULTIPLE_LINES",
+              encrypted_value:  "multiplelines",
+              secure: true
+            },
+            {
+              name:   "COMPLEX",
+              value:  "This has very <complex> data",
+              secure: false
+            }
+          ],
+          jobs:                    [],
+          approval:                {
+            type:          'manual',
+            authorization: {
+              users: [],
+              roles: []
+            }
+          }
+        }
+      ]
+    };
+
+    describe('deserialization', function () {
+      var template;
+
+      beforeAll(function() {
+        template = Template.fromJSON(templateJSON);
+      });
+
+      it('should initialize template model with name', function () {
+        expect(template.name()).toBe("template.name");
+      });
+
+      it('should initialize template model with stages', function () {
+        var expectedStages = Stages.fromJSON(templateJSON.stages);
+        var actualStages = template.stages();
+
+        expect(actualStages.countStage()).toBe(expectedStages.countStage());
+        expect(actualStages.firstStage().name()).toBe(expectedStages.firstStage().name());
+      });
+
+      it("should serialize to JSON", function () {
```

Comment:
Is serialization really required? Since it's read only, only the json will be deserialised to the template object. Not the other way around?